routing/dht/handlers.go

```diff
@@ -131,9 +131,10 @@ func (dht *IpfsDHT) handleFindPeer(ctx context.Context, p peer.ID, pmes *pb.Mess
 	resp := pb.NewMessage(pmes.GetType(), "", pmes.GetClusterLevel())
 	var closest []peer.ID
 
-	// if looking for self... special case where we send it on CloserPeers.
+	// if looking for self... special case where we are looking for connected peers.
+	// Used for FindPeersConnectedToPeer
 	if peer.ID(pmes.GetKey()) == dht.self {
-		closest = []peer.ID{dht.self}
+		closest = dht.peerstore.Peers()
```

Comment:
this isn't right-- the `peerstore` of the dht may contain data about peers we're no longer connected to, and is also shared with other subsystems. 

maybe we can grab the peers we're connected to from the routing table:

```go
dht.routingTable.ListPeers()
```

thoughts @whyrusleeping ?